docs/future/spa-mode.md

```diff
@@ -222,14 +222,41 @@ startTransition(() => {
 
 - You cannot call `serverLoader`/`serverAction` from your `clientLoader`/`clientAction` methods since there is no running server -- those will throw a runtime error if called
 
-- It's important to note that Remix SPA mode generates your `index.html` file by performing a "pre-render" of your root route on the server during the build
+### Server Build
```

Comment:
Moved the bottom 2 more complex bullet points to sub headings to make them linkable and facilitate a code snippet